docs/release-strategy.md

```diff
@@ -0,0 +1,69 @@
+# InstructLab CLI Release Strategy
+
+This document discusses the release strategy and processes for the
+`instructlab` Python package built from the
+<https://github.com/instructlab/instructlab> git repository.
+
+## Versioning Scheme
+
+Releases use a `X.Y.Z` numbering scheme.
+
+X-stream release are for major releases. At this stage in the project a major release has not been cut and we expect each release (2-3 weeks) to be a new Y-stream.
+
+Backwards compatibility will exist within a Y-stream for example within 0.1.8 and 0.1.9 and between Y-stream releases for update purposes. For example between 0.1.9 and 0.2.0.
```

Comment:
@cdoern I should probably make the wording say subsequent Y-stream releases right? We're not ensuring compatibility between 0.1.9 and 0.3.X but we are 0.2.9 and 0.3.X?

Reply:
I would take out the mention of update compatibility. I'm not totally sure what that means. I think you just mean you can `pip upgrade` or whatever? I'm not sure how to NOT allow that and I don't want anyone to think we're doing anything extra clever.

Reply:
@tiran Do you know if there are any expectations the python community has around `pip upgrade` or if you have any comments update compatibility in general?

Reply:
No, projects are free to define their own compatibility and update policy. People often assume CalVer or SemVer these days, but nobody is forcing projects to adopt SemVer or CalVer. Also we are < 1.0, so there is no backwards compatibility promise anyway.

Reply:
To @tiran's point about pre-1.0, I think I would remove the comment about compatibility of upgrades. If we say that, I think we need to get more detailed about what that means. For example, there's a proposal about completely changing the CLI subcommand layout (a good thing, I like it to be clear!) but that breaks compatibility from a user perspective. I don't think we're quite far enough along to require maintaining backwards compatibility with all of the old commands.